File: contactmanager/app/src/main/java/org/techtown/contactmanager/MainActivity.java
package org.techtown.contactmanager;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;

import org.techtown.contactmanager.data.DatabaseHandler;
import org.techtown.contactmanager.model.Contact;

import java.util.List;

public class MainActivity extends AppCompatActivity {

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);
    DatabaseHandler db = new DatabaseHandler(MainActivity.this);

    Log.d("Count", "onCreate: " + db.getCount());

    // create contact object first
    Contact jeremy = new Contact();
    jeremy.setName("Jeremy");
    jeremy.setPhoneNumber("987654321");

    Contact j = new Contact();
    j.setName("jason");
    j.setPhoneNumber("1234");

    db.addContact(jeremy);

//    Contact c = db.getContact(1);
//    c.setName("NewJeremy");
//    c.setPhoneNumber("2323");
//
//    int updateRow = db.updateContact(c);

    List<Contact> contactList = db.getAllContacts();

    for (Contact contact : contactList) {
      Log.d("MainActivity", "onCreate: " + contact.getName());
    }
  }
}

File: counter/app/src/main/java/org/techtown/counter/MainActivity.kt
package org.techtown.counter

import android.app.Activity
import android.app.Application
import android.os.Build
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import androidx.annotation.RequiresApi
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    companion object {
        val TAG = MainActivity::class.java.simpleName
    }

    var count = 0

    @RequiresApi(Build.VERSION_CODES.Q)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        add_button.setOnClickListener {
            count++
            counter_text.text = "$count"
        }

        sub_button.setOnClickListener {
            count--
            counter_text.text = "$count"
        }

//        registerActivityLifecycleCallbacks(object : Application.ActivityLifecycleCallbacks{
//            override fun onActivityPaused(p0: Activity) {
//                Log.d(TAG, "onActivityPaused: ")
//            }
//
//            override fun onActivityStarted(p0: Activity) {
//                Log.d(TAG, "onActivityStarted: ")
//            }
//
//            override fun onActivityDestroyed(p0: Activity) {
//                Log.d(TAG, "onActivityDestroyed: ")
//            }
//
//            override fun onActivitySaveInstanceState(p0: Activity, p1: Bundle) {
//                Log.d(TAG, "onActivitySaveInstanceState: ")
//            }
//
//            override fun onActivityStopped(p0: Activity) {
//                Log.d(TAG, "onActivityStopped: ")
//            }
//
//            override fun onActivityCreated(p0: Activity, p1: Bundle?) {
//                Log.d(TAG, "onActivityCreated: ")
//            }
//
//            override fun onActivityResumed(p0: Activity) {
//                Log.d(TAG, "onActivityResumed: ")
//            }
//
//        })
    }
}